test/docker-e2e/e2e_test.go

```diff
@@ -41,6 +44,61 @@ func (s *CelestiaTestSuite) SetupSuite() {
 	s.client, s.network = tastoradockertypes.DockerSetup(s.T())
 }
 
+// ExecuteNodeCommand executes a command on a chain node with common parameters automatically added.
+// This reduces boilerplate in tests by setting common flags like --home, --node, --fees, etc.
+func (s *CelestiaTestSuite) ExecuteNodeCommand(ctx context.Context, chainNode tastoratypes.ChainNode, cmd ...string) (string, string, error) {
+	s.Require().Greater(len(cmd), 0, "command must not be empty")
+
+	var finalCmd []string
+
+	if cmd[0] == "celestia-appd" {
+		s.Require().Greater(len(cmd), 1, "celestia-appd must have at least one subcommand")
+		cmd = cmd[1:]
+	}
+
+	isTxCommand := cmd[0] == "tx"
+	isKeysCommand := cmd[0] == "keys"
+
+	// Common flags for all commands
+	if !slices.Contains(cmd, "--home") {
+		finalCmd = append(finalCmd, "--home", homeDir)
+	}
+
+	// Add --node flag for all commands except key management commands
+	if !isKeysCommand && !slices.Contains(cmd, "--node") {
+		hostname, err := chainNode.GetInternalHostName(ctx)
+		if err != nil {
+			return "", "", err
+		}
+		finalCmd = append(finalCmd, "--node", fmt.Sprintf("tcp://%s:26657", hostname))
+	}
```

Comment:
[not strictly blocking but we should fix it imo]

this is a code smell to have all of these if statements 

just create the command correctly in the first place instead of adding all of this stuff in an opaque way

the builder pattern is helpful for this type of stuff

Reply:
Also a note, if we do end up just using existing rpcs, we won't need this function at all, it's currently used to interact with the signal module and query the upgrade module. All of this can be done over grpc